wp-modules/pattern-data-handlers/pattern-data-handlers.php

```diff
@@ -321,12 +322,9 @@ function update_pattern( $pattern ) {
  * @return bool Whether the deletion succeeded.
  */
 function delete_pattern( string $pattern_name ): bool {
-	$wp_filesystem = \PatternManager\GetWpFilesystem\get_wp_filesystem_api();
+	$wp_filesystem = get_wp_filesystem_api();
 	$pattern_path  = get_pattern_path( $pattern_name );
-	$result        = $wp_filesystem && $wp_filesystem->exists( $pattern_path ) && $wp_filesystem->delete( $pattern_path );
-	tree_shake_theme_images( $wp_filesystem, 'copy_dir' );
```

Comment:
This is probably the main cause of this PR's bug, though I can't reproduce it.

I shouldn't have called `tree_shake_theme_images()` in `delete_pattern()`.

It's not needed here. There are 2 other places that call it.

Reply:
Hiding it in `delete_pattern()` probably caused this bug, and invites further bugs.